lib/greenhouse_api/client.rb

```diff
@@ -98,18 +120,20 @@ def data_limit_reached?(data, limit)
       limit && data.length >= limit
     end
 
-    def headers
+    def headers(on_behalf_of = nil)
       credential = Base64.strict_encode64(@api_key + ':')
 
-      {
+      request_header = {
         'Authorization' => 'Basic ' + credential,
       }
+      request_header['On-Behalf-Of'] = on_behalf_of unless on_behalf_of.nil? || on_behalf_of.strip.empty?
+      request_header
     end
 
-    def request(http_method:, headers:, endpoint:, params:, body: {})
+    def request(http_method:, headers:, endpoint:, params: {}, body: {}, api_version: 'v1')
```

Comment:
I'm not sure if api_version should be configured per request like this. The first question I have is why do we need to switch to v2 for users api?

Secondly, if we do indeed need to specify api_version, do you think it's more appropriate to configure it in client initialization?

Reply:
The enable/disable user endpoints are only available via v2: https://developers.greenhouse.io/harvest.html?ruby#patch-disable-user
My guess is that these are endpoints they added recently? 
IMO it makes sense for us to specify v2 in the method. Requiring this upon client initialization makes it feel like the caller has to know more about the api than it should, wdyt?

Reply:
Your reasoning makes sense, reduce work on the consumers. Are the other endpoints (candidates, offers, etc.) available on v2. Should we migrate everything to v2?

Reply:
The other endpoints are only available on v1, I think because they were created in the first iteration of the api

Reply:
I see, your approach makes sense, the gem consumers shouldn't need to know about versions 👍